### PR TITLE
Translate ResourceValue to ResourceIr to add metadata

### PR DESCRIPTION
**Pull request by iph**
Resources in intermediate representation was the last holdover
to a working program without manual manipulation of typescript.

This change adds "Complexity", a concept to help translate CDK
interfaces appropriately for L1s, and adds this decoration to
ResourceIr. This change also adds a few other bells and whistles
along the way while debugging final output (like AWS::AccountId)

### CODE REVIEW

File: src/ir/resources.rs [hunk 1]
```diff
@@ -1 +1,297 @@
...
+    pub condition: Option<String>,
+    pub resource_type: String,
+    pub properties: HashMap<String, ResourceIr>,
+}
+
+pub fn translates_resources(parse_tree: &CloudformationParseTree) -> Vec<ResourceInstruction> {
+    let spec = spec();
+    let mut resource_instructions = Vec::new();
+    for resource in parse_tree.resources.resources.iter() {
+        let resource_spec = spec
```
Comment (roblaks) [inline src/ir/resources.rs:56]: Nit: This is more of an optimization question. Does this need to be done for every iteration? Can we save resource_spec after the first time we see the type?

File: src/ir/resources.rs [hunk 2]
```diff
@@ -1 +1,297 @@
...
+                            Complexity::Simple(_) => Option::None,
+                            Complexity::Complex(x) => {
+                                let mut ret =
+                                    format!("{}.{}", resource_translator.resource_type, x);
+                                if x == "Tag" {
+                                    ret = "Tag".to_string();
+                                }
+                                Option::Some(ret)
+                            }
+                        };
```
Comment (roblaks) [inline src/ir/resources.rs:152]: Worth moving to its own method since it's re-used in two places?

**Review by roblaks — APPROVED** [review]

File: src/ir/resources.rs (hunk 1 above)
Comment (iph) [reply src/ir/resources.rs:56]: I'm a bit confused, as each resource is unique, and the lookup is O(1) already (as it's backed by a hashmap in spec).

File: src/ir/resources.rs (hunk 2 above)
Comment (iph) [reply src/ir/resources.rs:152]: Yeah. I'ma push so we got a working application but will then update to clean this up.

**Pull request merged**